### ISSUE TITLE
Add Qt Background task

### ISSUE THREAD

**Issue by Xasin** [Xasin/TapBadge#2]
This is gonna be simple, but we'll definitely need to do it for any more serious application. 
In essence, we need to split the code into a background section (handling the BT SPP connection and updating between Badge and MQTT), and a foreground section (which will be disjoint from the background section, and will only use MQTT to fetch data).

This has lower priority and depends on #1 to make much sense.
